crates/aptos-api-tester/src/main.rs

```diff
@@ -0,0 +1,541 @@
+            .await?;
+    client.wait_for_transaction(&pending_txn).await?;
+
+    Ok(())
+}
+
+/// Helper function that gets back the result of the interaction.
+async fn get_message(client: &Client, address: AccountAddress) -> Option<String> {
+    let resource = match client
+        .get_account_resource(
+            address,
+            format!("{}::message::MessageHolder", address.to_hex_literal()).as_str(),
+        )
+        .await
+    {
+        Ok(response) => response.into_inner()?,
+        Err(_) => return None,
+    };
+
+    Some(resource.data.get("message")?.as_str()?.to_owned())
+}
+
+/// Tests module publishing and interaction. Checks that:
+///   - module data exists
+///   - can interact with module
+///   - resources reflect interaction
+async fn test_module(client: &Client, account: &mut LocalAccount) -> Result<(), TestFailure> {
+    // publish module
+    let blob = publish_module(client, account).await?;
+
+    // check module data
+    let response = client
+        .get_account_module(account.address(), "message")
+        .await?;
+
+    let expected_bytecode = &blob;
+    let actual_bytecode = &response.inner().bytecode;
+
+    if expected_bytecode != actual_bytecode {
+        return Err(TestFailure::Fail("wrong bytecode"));
+    }
+
+    // interact with module
+    let message = "test message";
+    set_message(client, account, message).await?;
+
+    // check that the message is sent
+    let expected_message = message.to_string();
+    let actual_message = match get_message(client, account.address()).await {
+        Some(message) => message,
+        None => {
+            return Err(TestFailure::Error(anyhow!(
+                "module interaction isn't reflected"
+            )))
+        },
+    };
+
+    if expected_message != actual_message {
+        return Err(TestFailure::Fail(
+            "module interaction isn't reflected correctly",
+        ));
+    }
+
+    Ok(())
+}
+
+async fn test_flows(client: Client, faucet_client: FaucetClient) -> Result<()> {
+    // create clients
+    let coin_client = CoinClient::new(&client);
+    let token_client = TokenClient::new(&client);
+
+    // create and fund account for tests
+    let mut giray = LocalAccount::generate(&mut rand::rngs::OsRng);
+    faucet_client.fund(giray.address(), 100_000_000).await?;
+    println!("{:?}", giray.address());
+
+    let mut giray2 = LocalAccount::generate(&mut rand::rngs::OsRng);
+    faucet_client.fund(giray2.address(), 100_000_000).await?;
+    println!("{:?}", giray2.address());
+
+    // Test new account creation and funding
+    // this test is critical to pass for the next tests
+    if handle_result("new account", test_newaccount(&client, &giray, 100_000_000))
+        .await
+        .is_err()
+    {
+        return Err(anyhow!("returning early because new account test failed"));
+    }
+
+    // Flow 1: Coin transfer
+    let _ = handle_result(
```

Comment:
Curious why are all the test results thrown away?

Reply:
this is because we need the return value for dependencies so it is returned but most of them aren't used